go/vt/sqlparser/sql.y

```diff
@@ -4468,6 +4471,7 @@ table_references:
 table_reference:
   table_factor
 | join_table
+| json_table
```

Comment:
This should be a table_factor, not a table_reference

Reply:
Make sure you have parse tests for using json_table in joins, in unions, in subqueries, etc.